mono/mini/mini-amd64.c

```diff
@@ -3548,111 +3540,12 @@ mono_amd64_emit_tls_get (guint8* code, int dreg, int tls_offset)
-	amd64_alu_reg_imm (code, X86_CMP, offset_reg, MAX_TEB_TLS_SLOTS);
-	more_than_64_slots = code;
-	amd64_branch8 (code, X86_CC_GE, 0, TRUE);
-
-	//TLS slot array, _TEB.TlsSlots, is at offset TEB_TLS_SLOTS_OFFSET and index is offset * 8 in Windows 64-bit _TEB structure.
-	amd64_shift_reg_imm (code, X86_SHL, offset_reg, 3);
-	amd64_alu_reg_imm (code, X86_ADD, offset_reg, TEB_TLS_SLOTS_OFFSET);
-
-	//TEB pointer is stored in GS segment register on Windows x64. TLS slot is located at calculated offset from that pointer.
-	x86_prefix (code, X86_GS_PREFIX);
-	amd64_mov_reg_membase (code, dreg, offset_reg, 0, sizeof (gpointer));
-		
-	tls_get_reg_done = code;
-	amd64_jump8 (code, 0);
-
-	amd64_patch (more_than_64_slots, code);
-
-	//TLS expansion slots, _TEB.TlsExpansionSlots, is at offset TEB_TLS_EXPANSION_SLOTS_OFFSET in Windows 64-bit _TEB structure.
-	x86_prefix (code, X86_GS_PREFIX);
-	amd64_mov_reg_mem (code, dreg, TEB_TLS_EXPANSION_SLOTS_OFFSET, sizeof (gpointer));
-	
-	//Check for NULL in _TEB.TlsExpansionSlots.
-	amd64_test_reg_reg (code, dreg, dreg);
-	empty_slot = code;
-	amd64_branch8 (code, X86_CC_EQ, 0, TRUE);
-	
-	//TLS expansion slots are at index offset into the expansion array.
-	//Calculate for the MAX_TEB_TLS_SLOTS offsets, since the interessting offset is offset_reg - MAX_TEB_TLS_SLOTS.
-	amd64_alu_reg_imm (code, X86_SUB, offset_reg, MAX_TEB_TLS_SLOTS);
-	amd64_shift_reg_imm (code, X86_SHL, offset_reg, 3);
-	
-	amd64_mov_reg_memindex (code, dreg, dreg, 0, offset_reg, 0, sizeof (gpointer));
-	
-	amd64_patch (empty_slot, code);
-	amd64_patch (tls_get_reg_done, code);
-
-	if (tmp_reg != -1)
-		amd64_pop_reg (code, tmp_reg);
-
-	return code;
-}
-
-#endif
-
-static guint8*
-emit_tls_get_reg (guint8* code, int dreg, int offset_reg)
-{
-	/* offset_reg contains a value translated by mono_arch_translate_tls_offset () */
-#ifdef TARGET_OSX
-	if (dreg != offset_reg)
-		amd64_mov_reg_reg (code, dreg, offset_reg, sizeof (mgreg_t));
-	amd64_prefix (code, X86_GS_PREFIX);
-	amd64_mov_reg_membase (code, dreg, dreg, 0, sizeof (mgreg_t));
-#elif defined(__linux__)
-	int tmpreg = -1;
-
-	if (dreg == offset_reg) {
-		/* Use a temporary reg by saving it to the redzone */
-		tmpreg = dreg == AMD64_RAX ? AMD64_RCX : AMD64_RAX;
-		amd64_mov_membase_reg (code, AMD64_RSP, -8, tmpreg, 8);
-		amd64_mov_reg_reg (code, tmpreg, offset_reg, sizeof (gpointer));
-		offset_reg = tmpreg;
-	}
-	x86_prefix (code, X86_FS_PREFIX);
-	amd64_mov_reg_mem (code, dreg, 0, 8);
-	amd64_mov_reg_memindex (code, dreg, dreg, 0, offset_reg, 0, 8);
-	if (tmpreg != -1)
-		amd64_mov_reg_membase (code, tmpreg, AMD64_RSP, -8, 8);
-#elif defined(TARGET_WIN32)
-	code = emit_tls_get_reg_windows (code, dreg, offset_reg);
-#else
-	g_assert_not_reached ();
-#endif
-	return code;
-}
-
 static guint8*
-amd64_emit_tls_set (guint8 *code, int sreg, int tls_offset)
+mono_amd64_emit_tls_set (guint8 *code, int sreg, int tls_offset)
```

Comment:
We're losing windows support. It's ok if we do it to simplify getting this reviewed and merged. But we should add it back after.